core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1055,6 +1073,21 @@ public synchronized void afterMerge(OnGoingMerge merge) {
                     deactivateThrottling();
                 }
             }
+            if (engineConfig.isFlushWhenLastMergeFinished() && indexWriter.hasPendingMerges() == false) {
+                // if we have no pending merges and we are supposed to flush once merges have finished
+                // we try to renew a sync commit which is the case when we are having a big merge after we
+                // are inactive. If that didn't work we go and do a real flush which is ok since it only doesn't work
+                // if we either have records in the translog or if we don't have a sync ID at all...
```

Comment:
Maybe also say here that the motivation to "flush after all merges finish and we are inactive indexing-wise" is to free up transient disk usage of the (presumably biggish) segments that were just merged?